File: HackerRank/10.Time Conversion/TimeConversion.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    horas = int(s[:2])
    minutos_segundos = s[2:8]
    meridiano = s[8:]
    if meridiano == "PM" and horas != 12:
        horas += 12
    elif meridiano == "AM" and horas == 12:
        horas = 0
    return "{:02d}{}".format(horas, minutos_segundos)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    fptr.close()

File: HackerRank/10.Time Conversion/TimeConversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'timeConversion' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string timeConversion(string s) {
    int horas = std::stoi(s.substr(0, 2));
    std::string minutosSegundos = s.substr(2, 6);
    std::string meridiano = s.substr(8);
    if(meridiano == "PM" && horas != 12) {
        horas += 12;
    } else if(meridiano == "AM" && horas == 12) {
        horas = 0;
    }
    std::stringstream ss;
    ss << std::setw(2) << std::setfill('0') << horas << minutosSegundos;
    return ss.str();
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
